encodings/decimal-byte-parts/src/decimal_byte_parts/compute/compare.rs

```diff
@@ -91,7 +91,7 @@ fn decimal_value_wrapper_to_primitive(
 
 fn decimal_value_to_primitive<P>(decimal_value: DecimalValue) -> Result<ScalarValue, Sign>
 where
-    P: NativePType + NumCast + Bounded + ToPrimitive,
+    P: IntegerPType + ToI256,
```

Comment:
Should this be `DecimalValueType`?
```suggestion
    P: DecimalValueType,
```

Reply:
Do you mean we should rename the type? `DecimalValueType` already exists in `vortex-scalar` and it is an enum

Reply:
I was thinking this should be named `DecimalValueType` (similar to NativePType), but that is taken

Reply:
To get this over the line I made an issue that we can address later: https://github.com/vortex-data/vortex/issues/4808